From 2aae756d3722c9b2026cd731ab473b1c35e3248a Mon Sep 17 00:00:00 2001
From: Petr Machata <petrm@mellanox.com>
Date: Thu, 28 Nov 2019 16:07:57 +0200
Subject: [PATCH 3/3] mlnx_qos: Add flush command to dscp2prio

This command allows removing all existing entries in one go.

Signed-off-by: Petr Machata <petrm@mellanox.com>
---
 ofed_scripts/utils/mlnx_qos | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/ofed_scripts/utils/mlnx_qos b/ofed_scripts/utils/mlnx_qos
index 1c3e38d..b56d1ef 100755
--- a/ofed_scripts/utils/mlnx_qos
+++ b/ofed_scripts/utils/mlnx_qos
@@ -99,8 +99,8 @@ parser.add_option("--trust", dest="trust",
                   help="set priority trust state to pcp or dscp")
 parser.add_option("--dscp2prio", dest="dscp2prio",
 		  action="append",
-                  help="set/del a (dscp,prio) mapping. Example 'set,30,2' maps dscp 30 to priority 2. " +
-                       "'del,30,2' deletes the mapping and resets the dscp 30 mapping back to the default setting priority 0.")
+		  help="set/del a (dscp,prio) mapping, or 'flush' to delete all. Example 'set,30,2' maps dscp 30 to priority 2. " +
+		       "'del,30,2' deletes the mapping and resets the dscp 30 mapping back to the default setting priority 0.")
 parser.add_option("--defltprio", dest="defltprio",
                   help="set/del default priority. Example 'set,2' sets default priority to 2. " +
                        "'del,2' deletes this setting.")
@@ -435,6 +435,9 @@ if options.dscp2prio and options.trust:
 try:
 	if options.dscp2prio:
 		for opt in options.dscp2prio:
+			if opt == "flush":
+				dscp2prio.append((opt, None, None))
+				continue
 
 			action, dscp, prio = opt.split(",")
 			dscp = int(dscp)
@@ -460,6 +463,13 @@ try:
 		elif action == "del":
 			if (not is_trust_sysfs):
 				ctrl.del_ieee_app(IEEE_8021QAZ_APP_SEL_DSCP,prio,dscp)
+		elif action == "flush":
+			if (not is_trust_sysfs):
+				appTable = ctrl.get_ieee_app_table()
+				for app in appTable.apps.values():
+					if app.selector != IEEE_8021QAZ_APP_SEL_DSCP:
+						continue
+					ctrl.del_ieee_app(app.selector, app.priority, app.protocol)
 
 	if (not is_trust_sysfs):
 		appTable = ctrl.get_ieee_app_table()
-- 
2.17.1

